mtp_api/apps/prison/views.py

```diff
@@ -32,3 +37,44 @@ def create(self, request, *args, **kwargs):
 
     def perform_create(self, serializer):
         serializer.save(created_by=self.request.user)
+
+
+class PrisonerValidityView(mixins.RetrieveModelMixin, viewsets.GenericViewSet):
+    queryset = PrisonerLocation.objects.all()
+    permission_classes = (
+        IsAuthenticated, SendMoneyClientIDPermissions,
+        ActionsBasedPermissions,
+    )
+    serializer_class = PrisonerValiditySerializer
+    lookup_url_kwarg = 'prisoner_reference'
+
+    re_reference = re.compile(r'^(?P<prisoner_number>[a-z]\d{4}[a-z]{2})-'
+                              r'(?P<prisoner_dob>\d{4}-\d{2}-\d{2})$', re.IGNORECASE)
+
+    @classmethod
+    def parse_reference(cls, reference):
```

Comment:
this was supposed to be a rough and ready first attempt. i was going to update this to a POST with two parameters in the future… but i suppose i could do it now, just for you